### PR TITLE
Fix auto configure listener names in broker config

### PR DESCRIPTION
**Pull request by ozangunalp**
Configure listeners only when no listeners config was given.

### CODE REVIEW

**Comment by ozangunalp** [comment]
@k-wall can you take a look?

**Comment by k-wall** [comment]
Apologies, I was working on #4 to address the issue I raised. I didn't see this one come in.

**Comment by ozangunalp** [comment]
:+1: I think both make sense together, I'll merge #4 and this one on top.

File: kafka-server/src/test/java/com/ozangunalp/kafka/server/BrokerConfigTest.java [hunk 1]
```diff
@@ -42,6 +41,26 @@ void testOverrideAdvertisedListeners() {
...
+        props.put(KafkaConfig.AdvertisedListenersProp(), "PLAINTEXT://:9092");
+        props.put(KafkaConfig.ProcessRolesProp(), "broker");
+        props.put(KafkaConfig.ListenersProp(), "BROKER://:9093,PLAINTEXT://:9092");
+        props.put(KafkaConfig.ListenerSecurityProtocolMapProp(), "BROKER:PLAINTEXT");
+        Properties properties = BrokerConfig.defaultCoreConfig(props, "", 9092, 9093, 9094, PLAINTEXT);
+        assertThat(properties).doesNotContainKey(KafkaConfig.ControllerListenerNamesProp());
+        assertThat(properties).doesNotContainKey(KafkaConfig.InterBrokerListenerNameProp());
+        assertThat(properties).doesNotContainKey(KafkaConfig.EarlyStartListenersProp());
+        assertThat(properties).containsEntry(KafkaConfig.BrokerIdProp(), "1");
+        assertThat(properties).containsEntry(KafkaConfig.QuorumVotersProp(), "1@:9094");
```
[the remark this answers is not in the document]
Comment (ozangunalp) [reply kafka-server/src/test/java/com/ozangunalp/kafka/server/BrokerConfigTest.java:58]: Indeed, I tried to make it a bit clearer.

**Pull request merged**